catalog/core/catalog-core-sourcemetricsplugin/src/main/java/ddf/catalog/metrics/source/SourceMetricsImpl.java

```diff
@@ -197,6 +206,15 @@ public QueryResponse process(QueryResponse input)
       for (Map.Entry<String, Integer> source : totalHitsPerSource.entrySet()) {
```

Comment:
✏️I would suggest pulling this loop out into a private `updateResultsMetrics` method to keep the `process` method more focused.